pkg/client/typed/dynamic/client.go

```diff
@@ -79,28 +84,54 @@ func NewClient(conf *restclient.Config) (*Client, error) {
 	return &Client{cl: cl}, nil
 }
 
-// Resource returns an API interface to the specified resource for
-// this client's group and version. If resource is not a namespaced
-// resource, then namespace is ignored.
+// Resource returns an API interface to the specified resource for this client's
+// group and version. If resource is not a namespaced resource, then namespace
+// is ignored.
 func (c *Client) Resource(resource *unversioned.APIResource, namespace string) *ResourceClient {
-	return &ResourceClient{
+	rc := &ResourceClient{
 		cl:       c.cl,
 		resource: resource,
 		ns:       namespace,
 	}
+	return rc
+}
+
+// ParameterCodec wraps a parameterCodec around the Client.
+func (c *Client) ParameterCodec(parameterCodec runtime.ParameterCodec) *ClientWithParameterCodec {
```

Comment:
A wrapper is fine, too-- basically the same :).

On Fri, Jun 3, 2016 at 10:42 AM, Chao Xu <notifications@github.com> wrote:

> In pkg/client/typed/dynamic/client.go
> <https://github.com/kubernetes/kubernetes/pull/26493#discussion_r65747184>
> :
>
> >  		cl:       c.cl,
> >  		resource: resource,
> >  		ns:       namespace,
> >  	}
> > +	return rc
> > +}
> > +
> > +// ParameterCodec wraps a parameterCodec around the Client.
> > +func (c *Client) ParameterCodec(parameterCodec runtime.ParameterCodec) *ClientWithParameterCodec {
>
> Make a copy sounds heavy, sort of defeated effort of the client pool.
>
> What's wrong with this wrapper approach?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26493/files/9e63bbb93387ea03634786063bfdf45197aaf47b#r65747184>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAnglkhzK9hVVrNAbQvQYmV0BdPZhWx_ks5qIGeHgaJpZM4IpLXf>
> .
>